frontend/apps/core-app/src/components/RecordEditor/FieldInput/MultiSelectFieldInput.component.tsx

```diff
@@ -1,7 +1,128 @@
 import * as React from 'react';
+import {
+  Pressable,
+  FormControl,
+  Input,
+  Modal,
+  Button,
+  HStack,
+  Checkbox,
+  ChevronDownIcon,
+  usePropsResolution,
+} from 'native-base';
+import { useFormContext, useController } from 'react-hook-form';
+import { FormDefinition } from 'core-api-client';
 
-type Props = {};
+type Props = {
+  formId: string;
+  field: FormDefinition['fields'][number];
+};
+
+export const MultiSelectFieldInput: React.FC<Props> = ({ formId, field }) => {
+  const [open, setOpen] = React.useState(false);
+  const [internalValues, setInternalValues] = React.useState<string[]>([]);
+
+  const { control } = useFormContext();
+
+  const {
+    field: { onChange, onBlur, value, ref },
+    fieldState: { error, invalid },
+  } = useController({
+    name: `${formId}.${field.id}`,
+    control,
+    rules: {}, // TODO Record validation
+  });
+
+  React.useEffect(() => {
+    if (value) {
+      setInternalValues(value);
+    }
+  }, [JSON.stringify(value)]);
+
+  const handleOpenModal = () => setOpen(true);
+
+  const handleCloseModal = (reset: boolean) => () => {
+    setOpen(false);
+    if (reset) {
+      setInternalValues(value || []);
+    }
+  };
+
+  const handleAdd = () => {
+    onChange(internalValues);
+    handleCloseModal(false)();
```

Comment:
this doesn't seem to work, the modal didn't close for me when I added a subrecord

Reply:
Wrong file, this is multiselect. But good catch, the sub form field was calling open rather than close.